csrgen.py

```diff
@@ -0,0 +1,190 @@
+"""
+config_template = """[ req ]
+    default_bits           = 2048
+    distinguished_name     = req_distinguished_name
+    prompt                 = no
+    req_extensions         = v3_req
+[ req_distinguished_name ]
+    {% if c is not none -%}
+        C                      = {{s}}
+    {%- endif %}
+    {%- if l is not none -%}
+        L                      = {{l}}
+    {%- endif %}
+    {%- if s is not none -%}
+        ST                     = {{s}}
+    {%- endif %}
+    {%- if o is not none -%}
+        O                      = {{o}}
+    {%- endif %}
+{{'    CN                     = {{ cn }}
+[ v3_req ]
+    subjectAltName          = @alt_names
+
+[alt_names]
+    DNS.1 = {{ cn }}
+    {%- for domain in sans %}
+        {%- if domain %}
+    DNS.{{ loop.index +1 }} = {{ domain }}
+        {%- endif %}
+    {%- endfor %}
+'}}
+"""
+
+
+def check_file(file):
+    try:
+        os.stat(file).st_size > 1
+    except:
+        return False
+    else:
+        return file
+
+
+def check_domain(input):
+    if check_file(input):
+        return input, True
+    else:
+        domain_template = re.compile(
+            "^(?=.{1,255}$)(?!-)[A-Za-z0-9\-]{1,63}(\.[A-Za-z0-9\-]{1,63})*\.?(?<!-)$")
+        if domain_template.match(input):
+            return input.lower()
+        else:
+            raise argparse.ArgumentError(
+                "Unable to determine if this is a domain or a file. Please check your input.")
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description=description,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    subparser = parser.add_subparsers(dest="command")
+
+    csr = subparser.add_parser("csr", help="Create CSR",
+                               formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    csr.add_argument('-k', "--key", type=check_file, help='Specify a key')
+    csr.add_argument('-d', "--domain", type=check_domain, required=True,
+                     help="Specify a domain or a file with list of domains")
+    csr.add_argument("-o", "--org", help="Organization Name")
+
+    config = subparser.add_parser("config", help="Create config file")
+    config.add_argument("-c", "--country", help="Country short code")
+    config.add_argument("-l", "--locality", help="Locality/City")
+    config.add_argument("-s", "--state", help="State Name")
+    config.add_argument("-o", "--org", help="Organization Name", required=True)
+
+    if parser.parse_args().command:
+        return parser.parse_args()
+    else:
+        parser.print_help()
+        exit()
+
+
+def gen_csr_config(args, config_file):
+    tm = env.get_template(config_file)
+    print(config_file)
+    SANS = list()
+    print("Generating CSR configuration..")
+    if type(args.domain) is tuple:
+        domain_file, is_file = args.domain
+        with open(domain_file) as fp:
+            domain = fp.readline()
```

Comment:
Suggest using `for domain in fp:` instead of this combo of `readline()` and a while-loop. Code ends up much simpler & easier to maintain.